### ISSUE TITLE
Allow for running E2E tests by users with reduced privileges

### ISSUE THREAD

**Issue by mgencur** [knative/client#1352]
### Feature request

It should be possible to run E2E tests by users without cluster-admin permissions. This will be useful in restricted environments where the container runtime is managed by someone else or the end user is simply given reduced permissions. It should be possible to verify the basic functionality of Knative Client even without cluster-admin permissions.

The current state is following. Some resources might not be created/applied with a regular user (say a user with "project admin" privileges):
* cannot run "oc adm policy add-scc-to-group anyuid" [here](https://github.com/openshift/knative-client/blob/release-v0.19.1/openshift/e2e-common.sh#L92)
* cannot run oc adm policy add-scc-to-group anyuid system:authenticated
* for each test, namespaces are created dynamically, kn client e2e TS creates a new namespace dynamically using [this code](https://github.com/openshift/knative-client/blob/release-v0.19.1/lib/test/integration.go#L227), i.e. "kubectl create ns", then deletes namespace with "kubectl delete ns" . This is already tracked in https://github.com/knative/client/issues/1304

If the above issues are worked-around, there are still failures:
5 tests fail
In general, they fail because the user does not have sufficient permissions for creating/patching resources:
```
TestSink
broker.go:31: error executing 'label namespace kne2etests6 eventing.knative.dev/injection=enabled': stderr: Error from server (Forbidden): namespaces "kne2etests6" is forbidden: User "mvinkler" cannot patch resource "namespaces" in API group "" in the namespace "kne2etests6"

TestSourceList
source_apiserver_test.go:117: assertion failed: error is not nil: stderr: Error from server (Forbidden): clusterroles.rbac.authorization.k8s.io is forbidden: User "mvinkler" cannot create resource "clusterroles" in API group "rbac.authorization.k8s.io" at the cluster scope

TestSourceListTypes
result_collector.go:75: ERROR: Error: missing apiVersion or kind; try GetObjectKind().SetGroupVersionKind() if you know the type

TestSourceApiServer
source_apiserver_test.go:117: assertion failed: error is not nil: stderr: Error from server (Forbidden): clusterroles.rbac.authorization.k8s.io is forbidden: User "mvinkler" cannot create resource "clusterroles" in API group "rbac.authorization.k8s.io" at the cluster scope

TestBrokerTrigger
broker.go:31: error executing 'label namespace kne2etests13 eventing.knative.dev/injection=enabled': stderr: Error from server (Forbidden): namespaces "kne2etests13" is forbidden: User "mvinkler" cannot patch resource "namespaces" in API group "" in the namespace "kne2etests13"
```
If "oc adm policy add-scc-to-group anyuid system:authenticated" is not set by cluster-admin in advance, there is one more test that fails:
```
TestServiceOptions (can be fixed with oc adm policy add-scc-to-group anyuid system:authenticated)
Message:               pods "svc6-zxzxy-2-deployment-56b8745ff9-" is forbidden: unable to validate against any security context constraint: [spec.containers[0].securityContext.runAsUser: Invalid value: 1001: must be in the ranges: [1001090000, 1001099999]]
--- FAIL: TestServiceOptions (784.14s)
```

### Use case

Verifying Knative Client functionality in restricted environments such as OpenShift Dedicated or IBM Code Engine.

**Comment by rhuss** [comment]
Thanks a lot for reporting those issue. While some of them might look like OpenShift specific one (like the `anyuid` issue), I think there are also other Knative environments out there that are locked down (like not being able to label a namespace), so we should fix them.

Actually I see the following sub-issues that we can tackle:

* Create the broke explicitly with e.g. `kn broker create` instead of using namespace labels for injecting a broker
* Fix #1304  
* Check why the ApiServerSource test needs a cluster role
* Check why listing sources test needs a cluster role
* Switch in the service option check to a potential higher number for runAs (wouldn't harm other systems and would help on OpenShift)

**Comment by mgencur** [comment]
Here are some findings:
TestSourceList, ApiServerSource - thes tests passl when they're executed by a project admin and there's a Role/RoleBinding used instead of ClusterRole/ClusterRoleBinding, I dont' see a particular reason to use ClusterRoles here
TestSourceListTypes - this test gathers a list of Sources and produces a YAML output, and the yaml includes a CustomResourceDefinitionList which is not accessible by a normal user (similar to CRDs)
